tools/cxbe/Main.cpp

```diff
@@ -54,6 +69,87 @@ int main(int argc, char *argv[])
         szXbeTitle[40] = '\0';
     }
 
+    // interpret title id
+    if(szXbeTitleID[0])
+    {
+        bool hex = true;
+        for(int i = 0; szXbeTitleID[i]; ++i)
+        {
+            if(szXbeTitleID[i] == '-')
+            {
+                hex = false;
+                break;
+            }
+        }
+        if(hex)
+        {
+            sscanf(szXbeTitleID, "%x", &dwTitleId);
+        }
+        else
+        {
+            char titlechar[2] = { 'C', 'X' };
+            unsigned titleno = 9999;
+            sscanf(szXbeTitleID, "%c%c-%u", &titlechar[0], &titlechar[1], &titleno);
+            if(titleno > 0xFFFF)
+            {
+                printf("WARNING: Title ID number too high (max is 65535)\n");
+                titleno = 0xFFFF;
+            }
+            dwTitleId = (titlechar[0] << 24) | (titlechar[1] << 16) | titleno;
+        }
+    }
+
+    // interpret region flags
+    if(szXbeRegions[0])
+    {
+        char c;
+        for(int i = 0; (c = szXbeRegions[i]); ++i)
+        {
+            switch(c)
+            {
+                case '-':;
+                    dwRegions = 0;
+                    goto breakfor;
+                case 'a':;
+                    dwRegions = XBEIMAGE_GAME_REGION_NA | XBEIMAGE_GAME_REGION_JAPAN |
+                                XBEIMAGE_GAME_REGION_RESTOFWORLD |
+                                XBEIMAGE_GAME_REGION_MANUFACTURING;
+                    goto breakfor;
+                case 'n':;
+                    dwRegions |= XBEIMAGE_GAME_REGION_NA;
+                    break;
+                case 'j':;
+                    dwRegions |= XBEIMAGE_GAME_REGION_JAPAN;
+                    break;
+                case 'w':;
+                    dwRegions |= XBEIMAGE_GAME_REGION_RESTOFWORLD;
+                    break;
+                case 'm':;
+                    dwRegions |= XBEIMAGE_GAME_REGION_MANUFACTURING;
+                    break;
+                default:;
+                    printf("WARNING: Invalid region char: %c\n", c);
+                    break;
+            }
+        }
+    breakfor:;
+    }
+    else
+    {
+        dwRegions = XBEIMAGE_GAME_REGION_NA | XBEIMAGE_GAME_REGION_JAPAN |
+                    XBEIMAGE_GAME_REGION_RESTOFWORLD | XBEIMAGE_GAME_REGION_MANUFACTURING;
```

Comment:
Rather than explicitly doing this if `REGION` was not provided, I think it would be better to assign the default value for `REGION`, so it *always* exists, then use the code above the `else`.